Content.Client/Lobby/UI/LobbyCharacterPreviewPanel.cs

```diff
@@ -32,13 +32,12 @@ public sealed class LobbyCharacterPreviewPanel : Control
         private readonly BoxContainer _viewBox;
         private readonly Label _unloaded;
 
-        public LobbyCharacterPreviewPanel(IEntityManager entityManager,
-            IClientPreferencesManager preferencesManager,
-            IPrototypeManager prototypeManager)
+        public LobbyCharacterPreviewPanel()
         {
-            _entMan = entityManager;
-            _preferencesManager = preferencesManager;
-            _prototypeManager = prototypeManager;
+            //This only runs once, if you cry about optimization fite me - Jezi
+            _entMan = IoCManager.Resolve<IEntityManager>();
+            _preferencesManager = IoCManager.Resolve<IClientPreferencesManager>();
+            _prototypeManager = IoCManager.Resolve<IPrototypeManager>();
```

Comment:
What do you recommend then? [Injecting] them? I can convert it over to use that method. The constructor needs to have no arguments otherwise it causes issues with the XAML. I'll look into migrating all that away from ioc manager

Reply:
acruid had this to say: https://discord.com/channels/310555209753690112/900426319433728030/952346212470759424